### PR TITLE
add accommodation updater

### PR DESCRIPTION
**Pull request by ayrat-molotkov**

### CODE REVIEW

File: HappyTravel.JuniperConnector.Updater/Service/AccommodationUpdater.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            Country = country,
+            Locality = locality,
+            Name = data.HotelName,
+            Coordinates = GetPoint(data?.Address?.Latitude, data?.Address?.Longitude),
+            Modified = _dateTimeProvider.UtcNow()
+        };
+
+
+        static Point GetPoint(string latitudeString, string longitudeString)
+            => double.TryParse(latitudeString, out double latitude) && double.TryParse(longitudeString, out double longitude)
```
Comment (suhan3z) [inline HappyTravel.JuniperConnector.Updater/Service/AccommodationUpdater.cs:89]: If we have errors while converting we definitely shouldnt suppress them, and at least log them

File: HappyTravel.JuniperConnector.Updater/Service/AccommodationUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        static (string, string) GetCountryAndLocality(string zoneCode, List<Zone> zones)
+        {
+            Zone zone;
+            var zoneHierarchy = new List<Zone>();            
+
+            do
+            {
+                zone = zones.Single(x => x.Code == zoneCode);
+                zoneHierarchy.Add(zone);
+                zoneCode = zone.ParentCode;
```
Comment (suhan3z) [inline HappyTravel.JuniperConnector.Updater/Service/AccommodationUpdater.cs:103]: What if we can't find zoneCode?

File: HappyTravel.JuniperConnector.Updater/Settings/UpdateMode.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace HappyTravel.JuniperConnector.Updater.Settings;
+
+public enum UpdateMode
+{
+    Full = 1,
+    Incremental = 2
```
Comment (suhan3z) [inline HappyTravel.JuniperConnector.Updater/Settings/UpdateMode.cs:6]: Do we have implementation for incremental update somewhere?

File: HappyTravel.JuniperConnector.Updater/Settings/AccommodationDataUpdateOptions.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿namespace HappyTravel.JuniperConnector.Updater.Settings;
+
+public class AccommodationDataUpdateOptions
+{
+    public UpdateMode UpdateMode { get; set; }
```
Comment (suhan3z) [inline HappyTravel.JuniperConnector.Updater/Settings/AccommodationDataUpdateOptions.cs:5]: We should have comments for all public methods and fields

File: HappyTravel.JuniperConnector.Updater/Settings/UpdateMode.cs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿namespace HappyTravel.JuniperConnector.Updater.Settings;
+
+public enum UpdateMode
```
Comment (suhan3z) [inline HappyTravel.JuniperConnector.Updater/Settings/UpdateMode.cs:3]: Same for comment here

File: HappyTravel.JuniperConnector.Updater/Settings/UpdateMode.cs (hunk 3 above)
Comment (ayrat-molotkov) [reply HappyTravel.JuniperConnector.Updater/Settings/UpdateMode.cs:6]: Yes, we have

**Review by suhan3z — APPROVED** [review]

**Pull request merged**